cf-agent/package_module.c

```diff
@@ -0,0 +1,1611 @@
+    }
+    
+    if (OpenSubDB(&db_updates, db_id, module_wrapper->package_module->name))
+    {
+        char package_key[strlen(info->name) + 4];
+
+        xsnprintf(package_key, sizeof(package_key),
+                "N<%s>", info->name);
+        
+        Log(LOG_LEVEL_DEBUG, "Looking for key in updates: %s", package_key);
+         
+        if (HasKeyDB(db_updates, package_key, sizeof(package_key)))
+        {
+            Log(LOG_LEVEL_DEBUG, "Found key in updates database");
+            
+            updates_list = SeqNew(3, FreePackageInfo);
+            size_t val_size =
+                    ValueSizeDB(db_updates, package_key, sizeof(package_key));
+            char buff[val_size + 1];
+            buff[val_size] = '\0';
+
+            ReadDB(db_updates, package_key, buff, val_size);
+            Seq* updates = SeqStringFromString(buff, '\n');
+            
+            for (int i = 0; i < SeqLength(updates); i++)
+            {
+                PackageInfo *package = calloc(1, sizeof(PackageInfo));
+
+                char *package_line = SeqAt(updates, i);
+                char version[strlen(package_line)];
+                char arch[strlen(package_line)];
+
+                if (sscanf(package_line, "V<%[^>]>A<%[^>]>", version, arch) == 2)
+                {
+                    package->version = SafeStringDuplicate(version);
+                    package->arch = SafeStringDuplicate(arch);
+                    SeqAppend(updates_list, package);
+                }
+                else
+                {
+                    Log(LOG_LEVEL_INFO, "Not able to parse available updates "
+                        "line: %s", package_line);
+                    /* Some error occurred while scanning package updates. */
+                    FreePackageInfo(package);
+                }
+            }
+        }
+        CloseDB(db_updates);
+    }
+    return updates_list;
+}
+
+PromiseResult RepoInstall(EvalContext *ctx,
+                          PackageInfo *package_info,
+                          const NewPackages *policy_data,
+                          const PackageModuleWrapper *wrapper,
+                          int is_in_cache,
+                          enum cfopaction action,
+                          bool *verified)
+{
+    Log(LOG_LEVEL_DEBUG, "Installing repo type package: %d", is_in_cache);
+    
+    /* Package is not present in cache. */
+    if (is_in_cache == 0)
+    {
+        const char *version = package_info->version;
+        if (package_info->version &&
+                StringSafeEqual(package_info->version, "latest"))
+        {
+            version = NULL;
+        }
+        if (action == cfa_warn || DONTDO)
+        {
+            Log(LOG_LEVEL_NOTICE, "Should install repo type package: %s",
+                package_info->name);
+            return PROMISE_RESULT_FAIL;
+        }
+        
+        *verified = false; /* Verification will be done elsewhere. */
```

Comment:
Saying *where* verification happens would be far better than saying it's "elsewhere" - see above.